test/e2e/specs/site-editor/block-removal.spec.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Site editor block removal prompt', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test.beforeEach( async ( { admin, editor } ) => {
+		await admin.visitSiteEditor( {
+			postId: 'emptytheme//index',
+			postType: 'wp_template',
+		} );
+		await editor.canvas.click( 'body' );
+	} );
+
+	test( 'should appear when attempting to remove Query Block', async ( {
+		page,
+	} ) => {
+		// Open and focus List View
+		await page.getByRole( 'button', { name: 'List View' } ).click();
+
+		// Select and try to remove Query Loop
+		await page.getByRole( 'link', { name: 'Query Loop' } ).click();
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect the block removal prompt to have appeared
+		await expect(
+			page.getByText( 'Query Loop displays a list of posts or pages.' )
+		).toBeVisible();
+	} );
+
+	test( 'should not appear when attempting to remove something else', async ( {
+		editor,
+		page,
+	} ) => {
+		// Open and focus List View
+		await page.getByRole( 'button', { name: 'List View' } ).click();
+
+		// Select Query Loop list item
+		await page.getByRole( 'link', { name: 'Query Loop' } ).click();
+
+		// Reveal its inner blocks in the list view
+		await page.keyboard.press( 'ArrowRight' );
+
+		// Select its Post Template inner block
+		await page.getByRole( 'link', { name: 'Post Template' } ).click();
+
+		// Expect to remove the Post Template with no prompts
+		expect( await getQueryBlockInnerBlocks( editor ) ).toHaveLength( 1 );
+		await page.keyboard.press( 'Backspace' );
+		expect( await getQueryBlockInnerBlocks( editor ) ).toHaveLength( 0 );
```

Comment:
Suggestion (non-blocking): The first assertion here is redundant. We already know the block is there; otherwise, the click event above will fail.

Maybe we should assert that the "Post Template" is no longer in Canvas. That will also make the assertion more explicit.

```js
await expect(
			editor.canvas
				.getByRole( 'document', {
					name: 'Block: Post Template',
				} )
).toBeHidden();
```

Reply:
> The first assertion here is redundant.

Yeah, you're right, I should remove it.

> Maybe we should assert that the "Post Template" is no longer in Canvas

Thanks! I thought of something like this, but was worried it would add unnecessary wait times (until `.toBeHidden()` times out), but I haven't been working with E2E testing, I'm out of the loop when it comes to conventions and concessions, so I'll happily do whatever you and others recommend.

Reply:
The assertion should be fast enough.

The new [`toBeAttached()`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-attached) assertion is better suited for this, but our Node/NPM versions are holding us back from using the latest Playwright :(